aea/mail/base.proto

```diff
@@ -7,4 +7,6 @@ message Envelope{
     string sender = 2;
     string protocol_id = 3;
     bytes message = 4;
+    string uri = 5;
+    bool uri_is_set = 6;
```

Comment:
Do we need this 6th field? A raw uri cannot be an empty string by definition, so the default "" would mean the uri is not set.

Reply:
In general yes; same issue as we discussed exists in proto3. 
But you are right. Since in this case the default value, i.e. empty string, in fact means the field is not set, we can drop it.